gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureEntity.java

```diff
@@ -140,13 +156,47 @@ public static Feature toFeature(FeatureEntity featureEntity, Project project) {
     }
 
     if (featureEntity.getPolygonVertices() != null) {
-      // TODO : Implement save to local db;
+      PolygonFeature.Builder builder =
+          PolygonFeature.builder().setVertices(stringToList(featureEntity.getPolygonVertices()));
+      fillFeature(builder, featureEntity, project);
+      return builder.build();
     }
 
     throw new LocalDataConsistencyException(
         "No geometry data found in feature " + featureEntity.getId());
   }
 
+  @Nullable
+  public static String listToString(ImmutableList<Point> vertices) {
+    if (vertices == null || vertices.isEmpty()) {
```

Comment:
The arg isn't nullable, so this null check can never be true.

Reply:
updated

Reply:
This code seems unchanged. Did you push your local changes?

Reply:
updated